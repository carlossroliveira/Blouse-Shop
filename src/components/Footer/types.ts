// -------------------------------------------------
// Types
// -------------------------------------------------
export interface IFooterProps {
  color: 'black' | 'white';
}
