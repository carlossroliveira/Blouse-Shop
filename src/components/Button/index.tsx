// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonProps } from './types';

export const Button = (props: ButtonProps) => {
  return (
    <ContainerSC>
      <ButtonSC role="button">{props.text}</ButtonSC>
    </ContainerSC>
  );
};
