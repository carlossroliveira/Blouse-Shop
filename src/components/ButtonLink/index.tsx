// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, LinkSC } from './buttonLinkStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonLinkProps } from './types';

export const ButtonLink = (props: ButtonLinkProps) => {
  return (
    <ContainerSC>
      <LinkSC to={`${props.path}`} onClick={props.onClick}>
        {props.text}
      </LinkSC>
    </ContainerSC>
  );
};
