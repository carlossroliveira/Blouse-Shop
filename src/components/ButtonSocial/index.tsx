// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ButtonSC } from './buttonSocialStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonSocialProps } from './types';

export const ButtonSocial = (props: ButtonSocialProps) => {
  return (
    <ButtonSC role="button">
      <>
        {props.icon}
        {props.text}
      </>
    </ButtonSC>
  );
};
