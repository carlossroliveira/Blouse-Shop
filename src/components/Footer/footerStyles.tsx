// -------------------------------------------------
// Packages
// -------------------------------------------------
import styled from 'styled-components';

export const ContainerSC = styled.section`
  position: absolute;
  right: 8rem;
`;

export const ParagraphSC = styled.p`
  color: ${({ theme }) => theme.color.primary};
  font-weight: 600;
  font-size: 20px;
`;
