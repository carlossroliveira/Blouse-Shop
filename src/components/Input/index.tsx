// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './inputStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { InputProps } from './types';

export const Input = (props: InputProps) => {
  return (
    <ContainerSC>
      <input
        type={props.type}
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
      />
    </ContainerSC>
  );
};
