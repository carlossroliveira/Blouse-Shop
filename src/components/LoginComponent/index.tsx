import React, { FormEvent, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';

export const LoginComponent = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const NAVIGATION = useNavigate();
  const auth = useAuth();

  const handleSubmita = async (event: FormEvent) => {
    event.preventDefault();

    console.log(email);
    console.log(password);
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  };

  /*  async function onFinish(event: any, email: string, password: string) {
    event.preventDefault();
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  } */

  return (
    <>
      <form onSubmit={handleSubmita}>
        <div>
          <input
            type="text"
            placeholder="email"
            value={email}
            onChange={(event) => setEmail(event.target.value)}
          />
        </div>
        <div>
          <input
            type="text"
            placeholder="Senha"
            value={password}
            onChange={(event) => setPassword(event.target.value)}
          />
        </div>
        <button type="submit">Logar</button>
      </form>
    </>
  );
};
