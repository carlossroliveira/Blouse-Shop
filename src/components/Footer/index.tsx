// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ParagraphSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { IFooterProps } from './types';

export const Footer = (props: IFooterProps) => {
  const currentYear: number = new Date().getFullYear();

  return (
    <ContainerSC>
      <ParagraphSC Color={props.color}>
        &copy; {currentYear} | Carlos Oliveira
      </ParagraphSC>
    </ContainerSC>
  );
};
