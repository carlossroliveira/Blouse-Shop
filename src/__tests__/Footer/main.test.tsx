// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Footer } from '../../components/Footer';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../../utils/withThemeComponent';

const LoginWithTheme = WithThemeComponent(Footer);

describe('Testing error footer', () => {
  it('Should render the text correctly', () => {
    render(<LoginWithTheme color={'black'} />);

    expect(screen.getByText(/Â© 2022 \| carlos oliveira/i)).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(<LoginWithTheme color={'black'} />);
    expect(render(<LoginWithTheme color={'black'} />)).toMatchSnapshot();
  });
});
