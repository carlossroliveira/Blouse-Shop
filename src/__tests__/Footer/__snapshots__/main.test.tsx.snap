// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing error footer component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: absolute;
  right: 8rem;
}

.c1 {
  font-weight: 600;
  font-size: 20px;
  color: #383838;
}

@media (max-width:975px) {
  .c0 {
    margin-top: 3rem;
    margin-right: -1rem;
  }
}

<body>
    <div>
      <section
        class="c0"
      >
        <p
          class="c1"
        >
          © 
          2022
           | Carlos Oliveira
        </p>
      </section>
    </div>
    <div>
      <section
        class="c0"
      >
        <p
          class="c1"
        >
          © 
          2022
           | Carlos Oliveira
        </p>
      </section>
    </div>
  </body>,
  "container": .c0 {
  position: absolute;
  right: 8rem;
}

.c1 {
  font-weight: 600;
  font-size: 20px;
  color: #383838;
}

@media (max-width:975px) {
  .c0 {
    margin-top: 3rem;
    margin-right: -1rem;
  }
}

<div>
    <section
      class="c0"
    >
      <p
        class="c1"
      >
        © 
        2022
         | Carlos Oliveira
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
